### ISSUE TITLE
main api for launching

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#541]
## ❓ Questions and Help

Last week @kne42 and I were able to onboard a bunch of new users at imageXD, and I couldn't help but feel some of our main api was still a little confusing. I know we talked about this a lot a couple months ago, but I wanted to see if anyone else was feeling similarly after getting napari out in the wild a little more.

Specifically I felt that people found confusing the distinction between `view` and `Viewer`, the jump to the `add_*` functions, and then the call signature of `view` which is slightly different then `add_image` (for example you can pass images with names as keyword arguments - `astronaut=image` in view but not in `add_image`).

Another thing that's come up is we've started adding more convenience wrapping - like `add_multichannel`, and more layer types like `add_surface`, but they are a little harder to get to / use then `view` as it requires instantiating that empty `Viewer`, which just isn't so nice.

So - my proposal, which I don't think we've discussed before is to drop `view` and bump all our `add_*` methods up to the top `napari` level and have them return the viewer, so you can now do
```python
from napari import add_image
viewer = add_image(astro)
```
and in one line you've shown an image
or you could do 
```python
from napari import add_surface
viewer = add_surface(mri_surf)
```
and in another line you've added a surface etc.
or you could do 
```python
import napari
viewer = napari.add_multichannel(multi_img)
viewer.add_point(pts)
```
and in two lines you've got multichannel images and points.

I think this will provide much easier access to our napari.

I think I'd leave `Viewer` on `napari` too so you could still do
```python
from napari import Viewer, add_surface
viewer = Viewer(title='new title')
viewer.add_surface(mri_surf)
```
for example, but we'd drop `view` entirely.

One final question then is what should `viewer.add_*` return? Right now it returns the layer. We could leave it like that, but then `napari.add_*` returns a viewer and `viewer.add_*` returns a layer which might be weird. We could have it return the viewer too, but why would you need the viewer back if you just did `viewer.add_*`, though it is the most consistent, or it could just return nothing.

What do people think about this proposal? @jni @kne42 @royerloic @AhmetCanSolak @kevinyamauchi @bryantChhun @shanaxel42 @freeman-lab

**Comment by AhmetCanSolak** [comment]
> 
> So - my proposal, which I don't think we've discussed before is to drop `view` and bump all our `add_*` methods up to the top `napari` level and have them return the viewer, so you can now do
> 
> ```python
> from napari import add_image
> viewer = add_image(astro)
> ```

I am absolutely up for dropping `view`.
But i think we should promote this api in the way below:
```python
import napari
viewer = napari.add_image(astro)
```

or:
```python
import napari as npr
viewer = npr.add_image(astro)
```
😄 

> 
> I think I'd leave `Viewer` on `napari` too so you could still do
> 
> ```python
> from napari import Viewer, add_surface
> viewer = Viewer(title='new title')
> viewer.add_surface(mri_surf)
> ```
> 

i would argue that we can even let people access viewer via `napari.viewer` as a singleton. I know this is conflicting with multiple viewer possibilities but why not considering multiple naparis instead multiple viewers? This could bring some Qt benefits as well.

> One final question then is what should `viewer.add_*` return? Right now it returns the layer. We could leave it like that, but then `napari.add_*` returns a viewer and `viewer.add_*` returns a layer which might be weird. We could have it return the viewer too, but why would you need the viewer back if you just did `viewer.add_*`, though it is the most consistent, or it could just return nothing.
> 

I believe we shouldn't return anything from `viewer.add_*` and `napari.viewer` should be the way to get viewer object. In that case `napari.add_image` can basically encapsulate `viewer.add_image` as an example.

**Comment by sofroniewn** [comment]
I’m wary of enforcing a single viewer, as in a jupyter notebook context it could lead to unexpected interactions between code cells.

For example i’d say two success calls to napari.add_image should open two separate viewers. Though I will say we have some bugs right with our ipython console when we have multiple viewers, and I’m curious about the potential Qt benefits of just having one viewer.

I agree though that promoting napari.add_image or npr.add_image etc is the way to go.

**Comment by sofroniewn** [comment]
This has been closed by #542
